From: Heinz Mauelshagen <heinzm@redhat.com>
Date: Thu, 14 May 2009 14:27:35 +0200
Subject: [md] dm raid45: don't clear the suspend flag on recovery
Message-id: 1242304055.28404.125.camel@o
O-Subject: [RHEL 5.4 PATCH] dm: raid45 target doesn't create parity as expected by dmraid (isw)
Bugzilla: 499406
RH-Acked-by: Jonathan Brassow <jbrassow@redhat.com>

RHEL5.4 device mapper: raid45 target

Jonathan Brassow pointed out in

"Re: [RHEL 5.4 PATCH] dm: raid45 target doesn't create parity as expected by dmraid (isw)
Wed, 13 May 2009 11:07:07 -0500 (18:07 CEST)",

that the RAID target wouldn't restart a partial recovery on resume,
because it clears the related flag on suspend of the mapped device.

The attached one liner, which didn't make it into my previous bz499406
related patch for some reason, fixes the issue.

Please ACK.

Heinz

diff --git a/drivers/md/dm-raid45.c b/drivers/md/dm-raid45.c
index 48dae62..0d2ac0e 100644
--- a/drivers/md/dm-raid45.c
+++ b/drivers/md/dm-raid45.c
@@ -4178,7 +4178,6 @@ static void raid_presuspend(struct dm_target *ti)
 	if (RSRecover(rs))
 		rh_stop_recovery(rs->recover.rh);
 
-	ClearRSRecover(rs);
 	cancel_delayed_work(&rs->io.ws_do_raid);
 	flush_workqueue(rs->io.wq);
 	wait_ios(rs);	/* Wait for completion of all ios being processed. */
